File: app/src/main/java/cc/smartcash/wallet/Adapters/WalletDialogAdapter.java
package cc.smartcash.wallet.Adapters;

import android.app.AlertDialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import cc.smartcash.wallet.Models.Wallet;
import cc.smartcash.wallet.R;
import cc.smartcash.wallet.ViewHolders.WalletDialogViewHolder;

public class WalletDialogAdapter extends RecyclerView.Adapter<WalletDialogViewHolder> {
    private Context context;
    private ArrayList<Wallet> wallets;
    private EditText txtAddress;
    private AlertDialog dialog;

    public WalletDialogAdapter(Context context, ArrayList<Wallet> wallets, EditText txtAddress, AlertDialog dialog) {
        this.txtAddress = txtAddress;
        this.context = context;
        this.wallets = wallets;
        this.dialog = dialog;
    }

    public void setItems(ArrayList<Wallet> wallets) {
        this.wallets = wallets;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public WalletDialogViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(context).inflate(R.layout.wallet_dialog_item, viewGroup, false);
        return new WalletDialogViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull WalletDialogViewHolder walletDialogViewHolder, int i) {
        walletDialogViewHolder.name.setText(String.valueOf(this.wallets.get(i).getDisplayName()));
        walletDialogViewHolder.btnSelect.setOnClickListener(v -> {
            this.txtAddress.setText(this.wallets.get(i).getAddress());
            dialog.hide();
        });
    }

    @Override
    public int getItemCount() {
        return wallets.size();
    }
}

File: app/src/main/java/cc/smartcash/smarthub/adapters/WalletDialogAdapter.kt
package cc.smartcash.smarthub.adapters

import android.app.AlertDialog
import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.EditText
import androidx.recyclerview.widget.RecyclerView
import cc.smartcash.smarthub.R
import cc.smartcash.smarthub.models.Wallet
import cc.smartcash.smarthub.viewHolders.WalletDialogViewHolder
import java.util.*

class WalletDialogAdapter(private val context: Context, private var wallets: ArrayList<Wallet>?, private val txtAddress: EditText, private val dialog: AlertDialog) : RecyclerView.Adapter<WalletDialogViewHolder>() {

    fun setItems(wallets: ArrayList<Wallet>) {
        this.wallets = wallets
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): WalletDialogViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.wallet_dialog_item, viewGroup, false)
        return WalletDialogViewHolder(view)
    }

    override fun onBindViewHolder(walletDialogViewHolder: WalletDialogViewHolder, i: Int) {
        walletDialogViewHolder.name.text = this.wallets!![i].displayName.toString()
        walletDialogViewHolder.btnSelect.setOnClickListener { v ->
            this.txtAddress.setText(this.wallets!![i].address)
            dialog.hide()
        }
    }

    override fun getItemCount(): Int {
        return wallets!!.size
    }
}
